fsw/cfe-core/src/sb/cfe_sb_priv.c

```diff
@@ -497,24 +497,33 @@ CFE_SB_RouteEntry_t* CFE_SB_GetRoutePtrFromIdx(CFE_SB_MsgRouteIdx_t RouteIdx)
 **    Pipe ID.
 **
 **  Arguments:
-**    MsgId  : ID of the message
+**    MsgId       : ID of the message
+**    PipeNameBuf : Buffer to receive name, must be OS_MAX_API_NAME bytes long
 **
 **  Return:
-**    Will return a pointer to the PipeName array in the pipe table if the the pipeid
-**    is in range. Otherwise this function returns a pointer to the
-**    PipeName4ErrCase[0], which is initialized with a null terminator.
+**    Will return CFE_SUCCESS and populate PipeNameBuf with the name
+**    of the pipe on success, otherwise returns CFE_SB_FAILED on error.
 **
 */
-char *CFE_SB_GetPipeName(CFE_SB_PipeId_t PipeId){
-
-    static char PipeName4ErrCase[1] = {'\0'};
+int32 CFE_SB_GetPipeName(CFE_SB_PipeId_t PipeId, char *PipeNameBuf){
```

Comment:
Is it worth checking to see if PipeNameBuf is non-NULL? If it's only used internally, then it might not be necessary.

Reply:
I would also prefer to see this have a third parameter, which is the size of the output buffer, rather than assuming it is always exactly OS_MAX_API_NAME.  There are several reasons for this, but mainly that OS_MAX_API_NAME is an OSAL definition, not a CFE platform definition, and theoretically a single build of CFE could be linked with a different build of OSAL without necessarily having the same definition.  

Also note that the `CFE_ES_GetAppName()` function, which is similar in concept/purpose, has the following prototype:
`int32 CFE_ES_GetAppName(char *AppName, uint32 AppId, uint32 BufferLength)`

The output/destination buffer pointer is first, which mirrors the way many C library string functions work.